web/commands/condor_status.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+import html
+import shlex
+import subprocess
+import time
+
+from flask import current_app, request
+
+from .. import formatting, slack
+from ..executor import executor
+from . import commands
+
+
+class CondorStatusCommandHandler(commands.CommandHandler):
+    def handle(self):
+        channel = request.form.get("channel_id")
+        user = request.form.get("user_id")
+        text = html.unescape(request.form.get("text"))
+
+        executor.submit(condor_status_reply, channel, user, text)
+
+        return f":thinking_face: :card_file_box:", 200
+
+
+def condor_status_reply(channel: str, user: str, text: str):
+    msg, file = generate_condor_status_reply(user, text)
+
+    info = slack.upload_file(
+        initial_comment=msg,
+        content=file,
+        filetype="text",
+        channels=channel,
+        title="condor_status at {}".format(datetime.datetime.utcnow()),
+    )
+
+    executor.submit(delete_file, info["file"]["id"])
+
+
+def delete_file(file_id: str, delay=datetime.timedelta(hours=6).total_seconds()):
+    time.sleep(delay)
+    slack.delete_file(file=file_id)
```

Comment:
Won't this leak files (wherever Slack is storing them) if the container or process is restarted, e.g., because of dependabot?

Part of my thinking for suggesting naming the files with their creation date was to make it easier to delete the old ones the next time we created a new one.  Turns out that `files.list()` includes a `created` field, so it should be easy to sweep old files.

Actually, we only really need to do the sweep the _first_ time we upload a new file in a given process/container.

Reply:
Yep, this will leak files on restarts. I just realized that `files.list` also lets you filter by users, so it should actually be easy to set up the initial sweep.

Reply:
Resolved, and also realized that putting sleeps in the executor threads was probably bad idea, so shifted that into yet another thread.